From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yive <6853318+Yive@users.noreply.github.com>
Date: Sun, 18 Feb 2024 01:35:04 -0800
Subject: [PATCH] Configuration for block ticks

Values below 1 will disable ticking the blockTicks list.
The list is used by stuff such as observers and other redstone components.

diff --git a/src/main/java/dev/yive/pluto/PlutoConfig.java b/src/main/java/dev/yive/pluto/PlutoConfig.java
index 8a7780669a09666a4b1fad3297ac2b8082f644fa..634e11d1f5a9353174caf7266bf38cb78063f878 100644
--- a/src/main/java/dev/yive/pluto/PlutoConfig.java
+++ b/src/main/java/dev/yive/pluto/PlutoConfig.java
@@ -352,5 +352,10 @@ public class PlutoConfig {
         private void disableFlapEvents() {
             disableFlapEvents = getBoolean("entities.global.disable-flapping-game-event", false);
         }
+
+        public int blockTickingModulo = 1;
+        private void blockTicking() {
+            blockTickingModulo = getInt("blocks.global.block-ticking-modulo", blockTickingModulo);
+        }
     }
 }
diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index a675cde06745eef00c92a2f9ace14f9190088999..d9482a9a595b84ed911bd9a79ca6d31c46d531de 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -670,6 +670,7 @@ public class ServerLevel extends Level implements WorldGenLevel, ca.spottedleaf.
         if (!this.isDebug() && flag) {
             j = this.getGameTime();
             gameprofilerfiller.push("blockTicks");
+            if (plutoConfig().blockTickingModulo > 0 && (plutoConfig().blockTickingModulo == 1 || MinecraftServer.currentTick % plutoConfig().blockTickingModulo == 0)) // Pluto - Configuration for block ticks
             this.blockTicks.tick(j, paperConfig().environment.maxBlockTicks, this::tickBlock); // Paper - configurable max block ticks
             gameprofilerfiller.popPush("fluidTicks");
             this.fluidTicks.tick(j, paperConfig().environment.maxFluidTicks, this::tickFluid); // Paper - configurable max fluid ticks
